keras/backend/tensorflow/rnn.py

```diff
@@ -457,15 +456,14 @@ def gru(
     time_major=False,
     reset_after=True,
 ):
-    inputs_supported = _do_rnn_inputs_support_cudnn(mask, time_major)
     cudnn_supported = cudnn_ok(
         activation,
         recurrent_activation,
         unroll,
         use_bias=bias is not None,
         reset_after=reset_after,
     )
-    if not cudnn_supported or not inputs_supported:
+    if not cudnn_supported:
```

Comment:
Let's do the safe thing and fallback to non-cudnn if there is any mask. Do `if not cudnn_supported or mask is not None:`

Reply:
Thanks. Done.  Also verified no regression in Colab test for both issues.